vsphere/config.go

```diff
@@ -205,10 +205,12 @@ func (c *Config) Client() (*Client, error) {
 	}
 
 	if isEligibleVSANEndpoint(client.vimClient) {
+		client.vimClient.UseServiceVersion("vsan")
```

Comment:
This function can error, please check it and return error if not nil

Reply:
Thanks for review, have addressed the comment.